### ISSUE TITLE
Bug in Sensor flush events

### ISSUE THREAD

**Issue by askthedragon** [carbonblack/cbapi-python#69]
```
In [11]: sensor.flush_events()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-11-3ad7638b7090> in <module>()
----> 1 sensor.flush_events()

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/cbapi/response/models.pyc in flush_events(self)
    726         :warning: This may cause a significant amount of network traffic from this sensor to the Cb Response Server
    727         """
--> 728         self.event_log_flush_time = datetime.now() + timedelta(days=1)
    729         self.save()
    730 

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/cbapi/models.pyc in __setattr__(self, attrname, val)
    316     def __setattr__(self, attrname, val):
    317         # allow subclasses to define their own property setters
--> 318         propobj = getattr(self.__class__, attrname, None)
    319         if isinstance(propobj, property) and propobj.fset:
    320             return propobj.fset(self, val)

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/cbapi/models.pyc in __get__(self, instance, instance_type)
    126     def __get__(self, instance, instance_type=None):
    127         d = super(DateTimeFieldDescriptor, self).__get__(instance, instance_type)
--> 128         return convert_from_cb(d)
    129 
    130     def __set__(self, instance, value):

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/cbapi/response/utils.pyc in convert_from_cb(s)
     53 def convert_from_cb(s):
     54     # Use dateutil.parser to parse incoming dates; flexible on what we receive, strict on what we send.
---> 55     return dateutil.parser.parse(s)
     56 
     57 

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/dateutil/parser.pyc in parse(timestr, parserinfo, **kwargs)
   1166         return parser(parserinfo).parse(timestr, **kwargs)
   1167     else:
-> 1168         return DEFAULTPARSER.parse(timestr, **kwargs)
   1169 
   1170 

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/dateutil/parser.pyc in parse(self, timestr, default, ignoretz, tzinfos, **kwargs)
    554             effective_dt = default
    555 
--> 556         res, skipped_tokens = self._parse(timestr, **kwargs)
    557 
    558         if res is None:

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/dateutil/parser.pyc in _parse(self, timestr, dayfirst, yearfirst, fuzzy, fuzzy_with_tokens)
    673 
    674         res = self._result()
--> 675         l = _timelex.split(timestr)         # Splits the timestr into tokens
    676 
    677         # keep up with the last token skipped so we can recombine

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/dateutil/parser.pyc in split(cls, s)
    190     @classmethod
    191     def split(cls, s):
--> 192         return list(cls(s))
    193 
    194     @classmethod

/home/builduser/.virtualenvs/cb-infoblox-connector/lib/python2.7/site-packages/dateutil/parser.pyc in __init__(self, instream)
     59         if getattr(instream, 'read', None) is None:
     60             raise TypeError('Parser must be a string or character stream, not '
---> 61                             '{itype}'.format(itype=instream.__class__.__name__))
     62 
     63         self.instream = instream

TypeError: Parser must be a string or character stream, not NoneType

```

**Comment by jgarman** [comment]
Two things at work here: the first problem is that for some reason, it's trying to "read" the `event_log_flush_time` attribute before writing to it... and in this case, the `event_log_flush_time` is None, so that's why you get this error.

Problem is, when you fix this, it still doesn't work, as 6.x for some reason changed the default date/time format for this specific use case. See error from `/var/log/cb/coreservices/debug.log` below:

```
ValueError: time data '2017-07-11 23:39:19.749203' does not match format '%a, %d %b %Y %H:%M:%S %Z'
```

**Comment by jgarman** [comment]
Will have to address this by formatting the time/date string properly depending on the Cb Response server version...

**Comment by jgarman** [comment]
Fixed in 4339e7fd8cce623b9a1ba286237692d750130cff
